Scripts/RemediationScripts/Remediate-EnableAADForKubernetesService.ps1

```diff
@@ -0,0 +1,523 @@
+
+# Examples:
+    To remediate:
+        1. To review the Kubernetes Services in a Subscription that will be remediated:
+           Enable-AADForKubernetes -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
+
+        2. To enable AAD in all Kubernetes Services in a Subscription:
+           Enable-AADForKubernetes -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck
+
+        3. To enable AAD in all Kubernetes Services in a Subscription, from a previously taken snapshot:
+           Enable-AADForKubernetes -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202201011212\EnableAADForKubernetesServices\KubernetesClusterWithAADDisabled.csv
+
+        To know more about the options supported by the remediation command, execute:
+        Get-Help Enable-AADForKubernetes -Detailed        
+###>
+
+function Setup-Prerequisites
+{
+    <#
+        .SYNOPSIS
+        Checks if the prerequisites are met, else, sets them up.
+
+        .DESCRIPTION
+        Checks if the prerequisites are met, else, sets them up.
+        Includes installing any required Azure modules.
+
+        .INPUTS
+        None. You cannot pipe objects to Setup-Prerequisites.
+
+        .OUTPUTS
+        None. Setup-Prerequisites does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Setup-Prerequisites
+
+        .LINK
+        None
+    #>
+
+    # List of required modules
+    $requiredModules = @("Az.Accounts", "Az.Aks", "Az.Resources")
+
+    Write-Host "Required modules: $($requiredModules -join ', ')" -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host "Checking if the required modules are present..."
+
+    $availableModules = $(Get-Module -ListAvailable $requiredModules -ErrorAction Stop)
+
+    # Check if the required modules are installed.
+    $requiredModules | ForEach-Object {
+        if ($availableModules.Name -notcontains $_)
+        {
+            Write-Host "Installing $($_) module..." -ForegroundColor $([Constants]::MessageType.Info)
+            Install-Module -Name $_ -Scope CurrentUser -Repository 'PSGallery' -ErrorAction Stop
+        }
+        else
+        {
+            Write-Host "$($_) module is present." -ForegroundColor $([Constants]::MessageType.Update)
+        }
+    }
+}
+
+function Enable-AADForKubernetes
+{
+    <#
+        .SYNOPSIS
+        Remediates 'Azure_KubernetesService_AuthN_Enabled_AAD' Control.
+
+        .DESCRIPTION
+        Remediates 'Azure_KubernetesService_AuthN_Enabled_AAD' Control.
+        AAD should be enabled in Kubernetes Service.
+        
+        .PARAMETER SubscriptionId
+        Specifies the ID of the Subscription to be remediated.
+        
+        .PARAMETER Force
+        Specifies a forceful remediation without any prompts.
```

Comment:
As we can't roll back changes done by this script, force switch should be avoided in this script. (Please discuss with team in next sync up).

Reply:
Done.